serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/web/AdminController.java

```diff
@@ -0,0 +1,152 @@
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/v1")
+@Slf4j
+public class AdminController {
+
+    private final ProcessService processService;
+    private final DocumentTypeService documentTypeService;
+
+    public AdminController(ProcessService processService, DocumentTypeService documentTypeService) {
+        this.processService = processService;
+        this.documentTypeService = documentTypeService;
+    }
+
+    @GetMapping(value = "/processes/{identifier:.+}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Process> getProcess(@PathVariable String identifier) {
+        Optional<Process> process = processService.findByIdentifier(identifier);
+        return process.isPresent()
+                ? ResponseEntity.ok(process.get())
+                : ResponseEntity.notFound().build();
+    }
+
+    @PostMapping(value = "/processes", consumes = MediaType.APPLICATION_JSON_VALUE)
+    @Transactional
+    public ResponseEntity addProcess(@RequestBody Process process) {
+        try {
+            Optional<Process> existingProcess = processService.findByIdentifier(process.getIdentifier());
+            if (existingProcess.isPresent()) {
+                return ResponseEntity.status(HttpStatus.CONFLICT).build();
+            }
+            List<DocumentType> persistedDoctypes = Lists.newArrayList();
+            for (DocumentType type : process.getDocumentTypes()) {
+                Optional<DocumentType> docFind = documentTypeService.findByIdentifier(type.getIdentifier());
+                if (!docFind.isPresent()) {
+                    persistedDoctypes.add(documentTypeService.add(type));
+                } else {
+                    persistedDoctypes.add(docFind.get());
+                }
+            }
+            process.setDocumentTypes(persistedDoctypes);
+            processService.add(process);
+            UriComponents uriComponents = UriComponentsBuilder.fromUriString("/processes")
+                    .path(process.getIdentifier())
+                    .build();
+            return ResponseEntity.created(uriComponents.toUri()).build();
+        } catch (Exception e) {
+            log.error("Exception during process save", e);
+            return ResponseEntity.badRequest().body(e.getMessage());
+        }
+    }
+
+    @DeleteMapping(value = "/processes/{identifier:.+}")
+    public ResponseEntity deleteProcess(@PathVariable String identifier) {
+        try {
+            Optional<Process> process = processService.findByIdentifier(identifier);
+            if (process.isPresent()) {
+                processService.delete(process.get());
+                return ResponseEntity.noContent().build();
+            }
+            return ResponseEntity.notFound().build();
+        } catch (Exception e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)